src/main/java/com/synopsys/integration/alert/web/config/JobConfigActions.java

```diff
@@ -0,0 +1,169 @@
+import com.synopsys.integration.exception.IntegrationException;
+
+@Component
+public class JobConfigActions {
+    private BaseConfigurationAccessor configurationAccessor;
+    private FieldModelProcessor fieldModelProcessor;
+    private ContentConverter contentConverter;
+
+    @Autowired
+    public JobConfigActions(final BaseConfigurationAccessor configurationAccessor, final FieldModelProcessor fieldModelProcessor, final ContentConverter contentConverter) {
+        this.configurationAccessor = configurationAccessor;
+        this.fieldModelProcessor = fieldModelProcessor;
+        this.contentConverter = contentConverter;
+    }
+
+    public boolean doesJobExist(UUID id) throws AlertDatabaseConstraintException {
+        return null != id && configurationAccessor.getJobById(id).isPresent();
+    }
+
+    public Optional<GroupedFieldModel> getJobById(UUID id) throws AlertDatabaseConstraintException {
+        final Optional<ConfigurationJobModel> jobConfiguration = configurationAccessor.getJobById(id);
+        if (jobConfiguration.isPresent()) {
+            final GroupedFieldModel groupedFieldModel = readGroupedConfiguration(jobConfiguration.get());
+            return Optional.of(groupedFieldModel);
+        }
+        return Optional.empty();
+    }
+
+    public List<GroupedFieldModel> getAllJobs() throws AlertDatabaseConstraintException {
+        final List<ConfigurationJobModel> allJobs = configurationAccessor.getAllJobs();
+        List<GroupedFieldModel> jobFieldModels = new LinkedList<>();
+        for (ConfigurationJobModel configurationJobModel : allJobs) {
+            GroupedFieldModel jobFieldModel = readGroupedConfiguration(configurationJobModel);
+            jobFieldModels.add(jobFieldModel);
+        }
+        return jobFieldModels;
+    }
+
+    public void deleteJobById(UUID id) throws AlertDatabaseConstraintException {
+        final Optional<ConfigurationJobModel> jobs = configurationAccessor.getJobById(id);
+        if (jobs.isPresent()) {
+            ConfigurationJobModel configurationJobModel = jobs.get();
+            for (ConfigurationModel configurationModel : configurationJobModel.getCopyOfConfigurations()) {
+                fieldModelProcessor.deleteFieldModel(configurationModel);
+            }
+            configurationAccessor.deleteJob(configurationJobModel.getJobId());
+        }
+    }
+
+    public GroupedFieldModel saveJob(GroupedFieldModel groupedFieldModel) throws AlertFieldException, AlertDatabaseConstraintException {
+        validateJob(groupedFieldModel);
+        Set<String> descriptorNames = new HashSet<>();
+        Set<ConfigurationFieldModel> configurationFieldModels = new HashSet<>();
+        for (FieldModel fieldModel : groupedFieldModel.getFieldModels()) {
+            descriptorNames.add(fieldModel.getDescriptorName());
+            final Collection<ConfigurationFieldModel> configurationFieldModelMap = fieldModelProcessor.saveFieldModel(fieldModel).values();
```

Comment:
consider renaming the variable configurationFieldModelMap.  It isn't a map it's a collection. configurationFieldModels would work.